io-amqp1_0-impl/src/main/java/org/apache/pulsar/ecosystem/io/amqp/AmqpSource.java

```diff
@@ -93,11 +93,11 @@ public void onMessage(Message message) {
                     record = new QpidJmsRecord(
                             Optional.empty(),
                             ByteBuffer.wrap(((JmsTextMessage) message).getText().getBytes(StandardCharsets.UTF_8)),
-                            message);
+                            (JmsMessage) message);
```

Comment:
```java
                if (config.isOnlyTextMessage()) {
                    JmsTextMessage jmsMessage = (JmsTextMessage) message;
                    ByteBuffer byteBuffer = ByteBuffer.wrap(jmsMessage.getText().getBytes(StandardCharsets.UTF_8));
                    record = new QpidJmsRecord(Optional.empty(), byteBuffer, jmsMessage);
                } else {
                    JmsMessage jmsMessage = (JmsMessage) message;
                    ByteBuf byteBuf = AmqpCodec.encodeMessage((AmqpJmsMessageFacade) (jmsMessage.getFacade()));
                    record = new QpidJmsRecord(Optional.empty(), byteBuf.nioBuffer(), jmsMessage);
                }
```

nit: reduce multiple in-place type coerce.

Reply:
I've processed the comment, it looks better this way.